statsforecast/models.py

```diff
@@ -17,55 +19,141 @@
 from .arima import auto_arima_f, forecast_arima, fitted_arima
 from .ets import ets_f, forecast_ets
 
-# Cell
-def auto_arima(X: np.ndarray, # time series
-               h: int, # forecasting horizon
-               future_xreg=None, #future regressors
-               fitted: bool=False, # fitted values
-               season_length: int=1, # season_length
-               approximation: bool=False, # approximation
-               level: Optional[Tuple[int]] = None # level
-    ) -> np.ndarray:
-    y = X[:, 0] if X.ndim == 2 else X
-    xreg = X[:, 1:] if (X.ndim == 2 and X.shape[1] > 1) else None
-    with np.errstate(invalid='ignore'):
-        mod = auto_arima_f(
-            y,
-            xreg=xreg,
-            period=season_length,
-            approximation=approximation,
-            allowmean=False, allowdrift=False #not implemented yet
-        )
-    fcst = forecast_arima(mod, h, xreg=future_xreg, level=level)
-    mean = fcst['mean']
-    if fitted:
-        return {'mean': mean, 'fitted': fitted_arima(mod)}
-    if level is None:
+# %% ../nbs/models.ipynb 6
+class _TS:
```

Comment:
Do we have sklearn dependencies?
I have been thinking about using the BaseEstimator to inherit sklearn utilities.

Are there efficiency implications between _TS vs BaseEstimator?